app/models/event.rb

```diff
@@ -14,4 +14,20 @@ def host_name
   def invited_name
     user.full_name
   end
+
+  def self.upcoming_events
+    where('date_time >= ?', Time.zone.now).order(date_time: :desc)
+  end
+
+  def self.upcoming_invited_events(user)
+    upcoming_events.where(user_id: user.id)
+  end
```

Comment:
Nice!  I'm wondering if we can just `delegate` the `.upcoming_events` method to the User model as well?  I think it would work and then the `upcoming_invited_events` and `past_invited_events` become redundant.  It will be a little tricky because there are two references to `events` in the User model: one is `events, through: :gym_memberships` and the other is `invited_events` (directly to the Event model), so on the delegation method, we may need to include `prefix: true` so we can distinguish between the two.  We can discuss this offline.